### PR TITLE
[issue-189] Feature: multi-language support

### LINKED ISSUE #189: Multi-language application support
- English
- Japanese
- Deutsche

We should persist this, and reload across browser refresh, unlike the current explorer.

Once the application design is understood, please advise @DeeJayElly where to add the components.

### PR DESCRIPTION
**Pull request by DominikGuzei**
This PR brings multi-language support, integrated with the new language switch component 🎉 

Desktop:

Mobile:
Bildschirmfoto 2020-02-29 um 01 53 12

The added i18n logic works as follows:

- implemented with https://www.i18next.com/
- All pages are localized under subpaths like `/[locale]/blocks` (e.g: `/de/blocks` etc.)
- The index route at `/` just redirects to one of the localized index pages, depending on:
  - The explicit user preference (if the user already chose a language in a previous session)
  - or the browser locale (if it is one of the supported ones)
  - or English as default.
- `<html lang>` is correctly set for each language
- Each language is exported into a separate sub-folder in the build directory (and only the localization strings of the respective language are embedded statically)
- A new component <LocalizedLink> is used to abstract the i18n logic away from simple links

Note: Internally the app uses normalized routing paths and doesn't have to be concerned about the current locale at all (that's handled by the navigation feature)

Closes #189

### CODE REVIEW

**Comment by a-rukin** [comment]
image
width of blocks is different here

**Comment by a-rukin** [comment]
By the way, we probably want to go one of these routes for the small language selection:
- center languages 
- use mono font, so it will take exactly the same space for two letters 

p.s. why not capital letters?

**Comment by DominikGuzei** [comment]
@a-rukin issues fixed 👍

File: source/features/i18n/translations/ja.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "pageTitle": "!!!Epoch | $t(productTitle)"
+  },
+  "error": {
+    "pageTitle": "!!!Error 404 | $t(productTitle)"
+  },
+  "index": {
+    "pageTitle": "!!!$t(productTitle)"
+  },
+  "locales": {
+    "de": "!!!German",
```
Comment (rhyslbw) [inline source/features/i18n/translations/ja.json]:
```suggestion
    "de": "!!!Deutsch",
```

File: next.config.js [hunk 2]
```diff
@@ -75,5 +76,28 @@ module.exports = withPlugins(
 
       return config;
     },
-  }
+    exportTrailingSlash: true,
+    /**
+     * Generate language specific pages under /[locale]/... paths e.g:
+     * /de/ leads to the the German landing page
```
Comment (rhyslbw) [inline next.config.js]:
```suggestion
     * /de/ leads to the the Deutsch landing page
```

File: next.config.js [hunk 3]
```diff
@@ -75,5 +76,28 @@ module.exports = withPlugins(
...
+     * /de/block leads to the German block page etc.
```
Comment (rhyslbw) [inline next.config.js]:
```suggestion
     * /de/block leads to the Deutsch block page etc.
```

File: source/features/i18n/translations/en.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "pageTitle": "Epoch | $t(productTitle)"
+  },
+  "error": {
+    "pageTitle": "Error 404 | $t(productTitle)"
+  },
+  "index": {
+    "pageTitle": "$t(productTitle)"
+  },
+  "locales": {
+    "de": "German",
```
Comment (rhyslbw) [inline source/features/i18n/translations/en.json]:
```suggestion
    "de": "Deutsch",
```

**Pull request merged**